packages/fx-core/src/component/generator/generator.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ContextV3 } from "@microsoft/teamsfx-api";
+import path from "path";
+import {
+  Component,
+  sendTelemetryEvent,
+  TelemetryEvent,
+  TelemetryProperty,
+} from "../../common/telemetry";
+import { SampleActionSeq, GenerateAction, TemplateActionSeq } from "./generateAction";
+import { GenerateContext } from "./generateContext";
+import {
+  genFileDataRenderReplaceFn,
+  genFileNameRenderReplaceFn,
+  getSampleInfoFromName,
+  getValidSampleDestination,
+  mergeReplaceMap,
+  sampleDefaultOnActionError,
+  templateDefaultOnActionError,
+} from "./utils";
+
+export class Generator {
+  public static async generateTemplate(
+    templateName: string,
+    language: string,
+    destinationPath: string,
+    ctx: ContextV3
+  ): Promise<void> {
+    const appName = ctx.projectSetting?.appName;
+    const projectId = ctx.projectSetting?.projectId;
+    const dataReplace = mergeReplaceMap(
```

Comment:
the only difference between dataReplace and nameReplace is projectId, why can't they share the same map?